lib/ansible/utils/collection_loader.py

```diff
@@ -116,6 +116,29 @@ def load_module(self, fullname):
 
         return mod
 
+    def get_source(self, fullname):
+        mod = sys.modules.get(fullname)
+        if not mod:
+            mod = self.load_module(fullname)
+
+        with open(to_bytes(mod.__file__), 'rb') as fd:
+            source = fd.read()
+
+        return source
+
+    def get_code(self, fullname):
+        return compile(source=self.get_source(fullname), filename=self.get_filename(fullname), mode='exec', flags=0, dont_inherit=True)
+
+    def is_package(self, fullname):
+        return self.get_filename(fullname).endswith('__init__.py')
+
+    def get_filename(self, fullname):
+        mod = sys.modules.get(fullname)
```

Comment:
maybe?
```suggestion
        mod = sys.modules.get(fullname) or self.load_module(fullname)
```

Reply:
Filename lookup (and most likely source lookup) need to work without loading the module. Otherwise failure to load the module will result in failure to get the filename or the source of the module.

This can result in problems such as: https://github.com/ansible/ansible/issues/64466

Reply:
Isn't it the alternate syntax for the next if-block? `or ` part is only evaluated if the first expression in the chain is falsy. Am I missing something? I'm just suggesting an alternate/shorter conditional syntax...

Reply:
@webknjaz Sorry for the confusion. My comment is unrelated to yours. It just happens to be regarding the same line of code.

Reply:
Oh.. If you want to add a separate comment in a PR, you can click on that `"+"` next to the line once again and GitHub will create a separate comment thread for you, right under this one. It's useful for tracking multiple separate things and keeping them distinct :)